internal/internal_event_handlers.go

```diff
@@ -782,15 +788,51 @@ func (wc *workflowEnvironmentImpl) SideEffect(f func() (*commonpb.Payloads, erro
 	wc.logger.Debug("SideEffect Marker added", tagSideEffectID, sideEffectID)
 }
 
+func (wc *workflowEnvironmentImpl) lookupMutableSideEffect(id string) (*commonpb.Payloads, bool) {
+	if payloadAtCallCount, ok := wc.mutableSideEffect[id]; ok {
+		currentCallCount := wc.mutableSideEffectCallCounter[id]
+		// Sort the calls
+		calls := make([]int, 0)
+		for k := range payloadAtCallCount {
+			calls = append(calls, k)
+		}
+		sort.Ints(calls)
+		// Find the most recent call at/before the current call count
+		var payload *commonpb.Payloads
+		var foundIndex int
+		for i := len(calls) - 1; i >= 0; i-- {
+			if calls[i] <= currentCallCount {
+				payload = payloadAtCallCount[calls[i]]
+				foundIndex = i
+				break
+			}
+		}
+		if payload == nil {
+			return nil, false
+		}
+		// Garbage collect old entries
+		// TODO(quinn) unclear if this should be done aggressively at WFT boundry
+		for i := 0; i < foundIndex; i++ {
+			delete(payloadAtCallCount, calls[i])
+		}
+
+		return payload, true
+	}
+	return nil, false
+}
+
 func (wc *workflowEnvironmentImpl) MutableSideEffect(id string, f func() interface{}, equals func(a, b interface{}) bool) converter.EncodedValue {
```

Comment:
Do we need to add some logic to `isCommandMatchEvent` to make sure the counters match? Or is that too strict of determinism checking.

Reply:
`MutableSideEffect` skip command match

Reply:
Ah, I see that now, thanks!